include/fmt/chrono.h

```diff
@@ -1091,21 +1105,51 @@ template <typename OutputIt, typename Char> class tm_writer {
     out_ = copy_str<Char>(std::begin(buf) + offset, std::end(buf), out_);
   }
 
-  void on_utc_offset() { format_localized('z'); }
+  void on_utc_offset() {
+#if defined(_WIN32)
+    _tzset();
```

Comment:
We shouldn't call _tzset or tzset because it changes global state.

Reply:
``_tzset`` / ``tzset`` is used internally in ``std::time_put``.

Without calling them, it is impossible to get the timezone correctly (if the timezone is not stored in ``struct std::tm``).

Windows: https://docs.microsoft.com/en-us/cpp/c-runtime-library/daylight-dstbias-timezone-and-tzname?view=msvc-140 (Remarks section)
Solaris: https://docs.oracle.com/cd/E86824_01/html/E54766/ctime-3c.html

On Solaris and Windows tzset is threadsafe.

Reply:
@vitaut, What do you think about this?

Reply:
I think we should limit system-specific logic here and suggest just detecting if `tm_gmtoff` is available and otherwise go through `format_localized` as before. The presence of `tm_gmtoff` can be detected in a portable way using SFINAE.